src/ui/qt/workarea/HexView.cpp

```diff
@@ -0,0 +1,713 @@
+      newOffset = selectedOffset - BYTES_PER_LINE;
+      goto selectNewOffset;
+
+    case Qt::Key_Down: {
+      int selectedCol = (selectedOffset - vgmfile->dwOffset) % BYTES_PER_LINE;
+      int endOffset = selectedItem->dwOffset - vgmfile->dwOffset + selectedItem->unLength;
+      int itemEndCol = endOffset % BYTES_PER_LINE;
+      int itemEndLine = endOffset / BYTES_PER_LINE;
+      newOffset = vgmfile->dwOffset + ((itemEndLine + (selectedCol > itemEndCol ? 0 : 1)) * BYTES_PER_LINE) + selectedCol;
+      goto selectNewOffset;
+    }
+
+    case Qt::Key_Left:
+      newOffset = selectedItem->dwOffset - 1;
+      goto selectNewOffset;
+
+    case Qt::Key_Right:
+      newOffset = selectedItem->dwOffset + selectedItem->unLength;
+      goto selectNewOffset;
+
+    case Qt::Key_Escape:
+      selectionChanged(nullptr);
+      break;
+
+    selectNewOffset:
+      if (newOffset >= vgmfile->dwOffset && newOffset < (vgmfile->dwOffset + vgmfile->unLength)) {
+        selectedOffset = newOffset;
+        auto item = vgmfile->GetItemFromOffset(newOffset, false);
+        if (item) {
+          selectionChanged(item);
+        }
+      }
+      break;
+    default:
+      QWidget::keyPressEvent(event);  // Pass event up to base class, if not handled here
+  }
+}
+
+bool HexView::handleOverlayPaintEvent(QObject* obj, QEvent* event) {
+  if (event->type() == QEvent::Paint) {
+    auto overlay = qobject_cast<QWidget*>(obj);
+
+    QPainter painter(static_cast<QWidget*>(obj));
+    painter.fillRect(QRect(0, 0, BYTES_PER_LINE * 3 * charWidth, overlay->height()),
+                     QColor(0, 0, 0, 100));
+
+    painter.fillRect(QRect(((BYTES_PER_LINE * 3) + HEX_TO_ASCII_SPACING_CHARS) * charWidth + (charWidth / 2),
+                           0,
+                           BYTES_PER_LINE * charWidth, overlay->height()),
+                           QColor(0, 0, 0, 100));
+
+    return true;
+  }
+  return false;
+}
+
+bool HexView::handleSelectedItemPaintEvent(QObject* obj, QEvent* event) {
+  if (event->type() == QEvent::Paint) {
+    if (!selectedItem) {
+      return true;
+    }
+    auto widget = static_cast<QWidget*>(obj);
+    if (prevSelectedItem != selectedItem) {
+
+      prevSelectedItem = selectedItem;
+
+      qreal dpr = devicePixelRatioF();
+      auto widgetSize = widget->size();
+      auto pixmap = QPixmap(widgetSize.width() * dpr, widgetSize.height() * dpr);
+      pixmap.setDevicePixelRatio(dpr);
+      pixmap.fill(Qt::transparent);
+
+      QPainter pixmapPainter = QPainter(&pixmap);
+
+      int baseOffset = static_cast<int>(selectedItem->dwOffset - vgmfile->dwOffset);
+      int startLine = baseOffset / BYTES_PER_LINE;
+      int startColumn = baseOffset % BYTES_PER_LINE;
+      int numLines = ((startColumn + static_cast<int>(selectedItem->unLength)) / BYTES_PER_LINE) + 1;
+
+      uint8_t* data = new uint8_t[selectedItem->unLength];
+      vgmfile->GetBytes(selectedItem->dwOffset, selectedItem->unLength, data);
```

Comment:
```suggestion
      auto itemData = std::vector<uint8_t>(selectedItem->unLength):
      vgmfile->GetBytes(selectedItem->dwOffset, selectedItem->unLength, itemData.data());
```
This will increase performance as most of the items are small and would benefit from Small Buffer Optimization